sdcm/logcollector.py

```diff
@@ -0,0 +1,819 @@
+                    CommandLogEntity(name='scylla.yaml',
+                                     command='cat /etc/scylla/scylla.yaml'),
+                    CommandLogEntity(name='coredumps.info',
+                                     command='sudo coredumpctl info')
+                    ]
+    cluster_log_type = "db-cluster"
+
+
+class LoaderLogCollector(LogCollector):
+    cluster_log_type = "loader-set"
+
+
+class MonitorLogCollector(LogCollector):
+    """SCT monitor set log collector
+
+    Collect required log entites for Monitor set nodes
+
+    Extends:
+        LogCollector
+
+    """
+    log_entities = [
+        CommandLogEntity(name='aprom.log',
+                         command='docker logs --details -t aprom'),
+        CommandLogEntity(name='agraf.log',
+                         command='docker logs --details -t agraf'),
+        CommandLogEntity(name='aalert.log',
+                         command='docker logs --details -t aalert'),
+        FileLogEntity(name='scylla_manager.log',
+                      command='sudo journalctl -u scylla-manager.service --no-tail',
+                      search_locally=True),
+        PrometheusSnapshotsEntity(name='prometheus_data'),
+        MonitoringStackEntity(name='monitoring-stack'),
+        GrafanaScreenShotEntity(name='grafana-screenshot'),
+        GrafanaSnapshotEntity(name='grafana-snapshot')
+    ]
+    cluster_log_type = "monitor-set"
+
+
+class SCTLogCollector(LogCollector):
+    """logs for hydra test run
+
+    Find and collect local version of files
+    for sct
+
+    Extends:
+        LogCollector
+
+    """
+    log_entities = [
+        FileLogEntity(name='sct.log',
+                      search_locally=True),
+        FileLogEntity(name='raw_events.log',
+                      search_locally=True),
+        FileLogEntity(name='events.log',
+                      search_locally=True),
+        FileLogEntity(name='output.log',
+                      search_locally=True),
+        FileLogEntity(name='critical.log',
+                      search_locally=True)
+    ]
+    cluster_log_type = 'sct-runner'
+
+    def collect_logs(self, local_search_path=None):
+        for ent in self.log_entities:
+            ent.collect(None, self.local_dir, None, local_search_path=local_search_path)
+        # TODO: rewrite to zipfile module with zip_64
+        if not os.listdir(self.local_dir):
+            LOGGER.warning('No any local files')
+            return self.local_dir, None, None
+        final_archive = shutil.make_archive("{}".format(self.local_dir), "zip", root_dir=self.local_dir)
+
+        s3_link = self.upload_logs(final_archive, "{0.test_id}/{0.current_run}".format(self))
+        return self.local_dir, final_archive, s3_link
+
+
+class Collector:
+    """Collector instance
+
+    Collector instance which should be run to collect logs and additional info
+    as separate stage in pipeline on as subcommand in hydra
+
+    """
+    LOCAL_SEARCH_FOR_USER = "abykov"
```

Comment:
I don't understand what's this is for ? ... you are trying to identify if we are in JENKINS run or not ?

Reply:
Yes, you right. Just forgot to change to jenkins

Reply:
I think we want the to work from everywhere and not just from Jenkins workers. 

We should identify the worker, and copy needed files out of it, workers can be tagged in a way we could list them, and we can scan them for the specific test id

Reply:
Adding scan for test id on builders. but list of builders is hardcoded.